app/models/concerns/oregon_digital/generic_metadata.rb

```diff
@@ -601,52 +601,23 @@ module GenericMetadata
-      accepts_nested_attributes_for :editor, reject_if: id_blank, allow_destroy: true
-      accepts_nested_attributes_for :ethnographic_term, reject_if: id_blank, allow_destroy: true
-      accepts_nested_attributes_for :family, reject_if: id_blank, allow_destroy: true
-      accepts_nested_attributes_for :format, reject_if: id_blank, allow_destroy: true
-      accepts_nested_attributes_for :genus, reject_if: id_blank, allow_destroy: true
-      accepts_nested_attributes_for :illustrator, reject_if: id_blank, allow_destroy: true
-      accepts_nested_attributes_for :interviewee, reject_if: id_blank, allow_destroy: true
-      accepts_nested_attributes_for :interviewer, reject_if: id_blank, allow_destroy: true
-      accepts_nested_attributes_for :local_collection_name, reject_if: id_blank, allow_destroy: true
-      accepts_nested_attributes_for :lyricist, reject_if: id_blank, allow_destroy: true
-      accepts_nested_attributes_for :military_branch, reject_if: id_blank, allow_destroy: true
-      accepts_nested_attributes_for :order, reject_if: id_blank, allow_destroy: true
-      accepts_nested_attributes_for :owner, reject_if: id_blank, allow_destroy: true
-      accepts_nested_attributes_for :patron, reject_if: id_blank, allow_destroy: true
-      accepts_nested_attributes_for :photographer, reject_if: id_blank, allow_destroy: true
-      accepts_nested_attributes_for :phylum_or_division, reject_if: id_blank, allow_destroy: true
-      accepts_nested_attributes_for :print_maker, reject_if: id_blank, allow_destroy: true
-      accepts_nested_attributes_for :publisher, reject_if: id_blank, allow_destroy: true
-      accepts_nested_attributes_for :recipient, reject_if: id_blank, allow_destroy: true
-      accepts_nested_attributes_for :repository, reject_if: id_blank, allow_destroy: true
-      accepts_nested_attributes_for :species, reject_if: id_blank, allow_destroy: true
-      accepts_nested_attributes_for :subject, reject_if: id_blank, allow_destroy: true
-      accepts_nested_attributes_for :taxon_class, reject_if: id_blank, allow_destroy: true
-      accepts_nested_attributes_for :tgn, reject_if: id_blank, allow_destroy: true
-      accepts_nested_attributes_for :transcriber, reject_if: id_blank, allow_destroy: true
-      accepts_nested_attributes_for :translator, reject_if: id_blank, allow_destroy: true
-      accepts_nested_attributes_for :cultural_context, reject_if: id_blank, allow_destroy: true
-      accepts_nested_attributes_for :style_or_period, reject_if: id_blank, allow_destroy: true
-      accepts_nested_attributes_for :form_of_work, reject_if: id_blank, allow_destroy: true
-      accepts_nested_attributes_for :institution, reject_if: id_blank, allow_destroy: true
-      accepts_nested_attributes_for :workType, reject_if: id_blank, allow_destroy: true
+
+      # Sets controlled values
+      self.controlled_properties = properties.reject { |_k, v| v.class_name.nil? }.keys.map(&:to_sym)
```

Comment:
The assumption here is that any property with a `class_name` must be a controlled vocabulary.. it would be better if this checked for a class_name and that the class name includes `ControlledVocabularies`?

Reply:
As stated above, the class_name gets populated with the classes that are associated with a property. Which SHOULD only be controlled vocabularies, but it could change.